### PR TITLE
14159 Allow corrections to correction and conversion filings

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14159

*Description of changes:*
- updated correctThisFiling() to handle other filing types
- updated disableCorrection() to handle other filing types
- updated doResumeFiling() to handle other filing types
- refactored disableCorrection() to better line up with unit test
- updated unit test
- app version = 5.7.15

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -703,50 +703,22 @@ export default class FilingHistoryList extends Vue {
...
-          params: { correctedFilingId }
-        })
-        break
-
-      case FilingTypes.CHANGE_OF_ADDRESS:
-        this.$router.push({
-          name: Routes.CORRECTION,
-          params: { correctedFilingId }
-        })
-        break
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:728]: Now part of default block below.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 2]
```diff
@@ -898,18 +870,22 @@ export default class FilingHistoryList extends Vue {
...
+    conditions[3] = () => item.isFutureEffective
+    conditions[4] = () => this.isTypeAlteration(item)
+    conditions[5] = () => this.isTypeTransition(item)
+    conditions[6] = () => (this.isTypeIncorporationApplication(item) && !this.isBComp)
+    conditions[7] = () => (this.isTypeChangeOfRegistration(item) && !this.isFirm)
+    conditions[8] = () => (this.isTypeConversion(item) && !this.isFirm)
+    conditions[9] = () => (this.isTypeCorrection(item) && !this.isFirm && !this.isBComp)
+    conditions[10] = () => (this.isTypeRegistration(item) && !this.isFirm)
+
+    return conditions.some(condition => condition())
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:888]: This iterates over the functions and runs each of them until one returns True. (So it doesn't do more work than needed.)

I thought I'd structure this code like this to better line up with the unit test (which verifies each condition separately). It should also be easier to add conditions here. What do you think?

File: src/components/Dashboard/TodoList.vue [hunk 3]
```diff
@@ -1367,20 +1367,25 @@ export default class TodoList extends Vue {
...
+            break
+
+          default:
+            // resume local correction for all other filings
+            this.setCurrentFilingStatus(FilingStatus.DRAFT)
+            this.$router.push({
+              name: Routes.CORRECTION,
+              params: { filingId: item.filingId.toString(), correctedFilingId: item.correctedFilingId.toString() }
+            })
+            break
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:1388]: I structured this to look like correctThisFiling() in Filing History List.

**Comment by severinbeauvais** [comment]
Looks like GH is having issues today and not running CI tasks (checks). Or maybe something changed on our end.
cc: @pwei1018

**Comment by severinbeauvais** [comment]
> Looks like GH is having issues today and not running CI tasks (checks). Or maybe something changed on our end that I don't know about.
> 
> cc: @pwei1018

I squashed the 2 commits and force-pushed from local to re-trigger the CI action. It's working this time.

File: src/components/Dashboard/FilingHistoryList.vue (hunk 2 above)
Comment (lambert-alex) [reply src/components/Dashboard/FilingHistoryList.vue:888]: It looks pretty clean.

**Review by lambert-alex — APPROVED** [review]
LGTM

**Review by lewischenstudio — APPROVED** [review]
LGTM

**Review by ketaki-deodhar — APPROVED** [review]
Looks good

**Pull request merged**